.github/workflows/build.yaml

```diff
@@ -28,20 +47,33 @@ jobs:
           go-version: '1.19'
           cache: true
 
+      - name: Run GoReleaser build
+        uses: goreleaser/goreleaser-action@v4
+        if: ${{ github.event_name == 'pull_request' }}
+        with:
+          distribution: goreleaser
+          version: latest
+          args: release --snapshot
+
       - name: Run GoReleaser build and publish tags
-        uses: goreleaser/goreleaser-action@v3
+        uses: goreleaser/goreleaser-action@v4
+        if: startsWith(github.ref, 'refs/tags/')
         with:
           distribution: goreleaser
           version: latest
-          args: release --rm-dist
+          args: release --clean
```

Comment:
The --clean flag is not released yet, it'll be in v1.15 that I plan to release over the weekend